validphys2/src/validphys/covmats.py

```diff
@@ -736,6 +738,16 @@ def pdferr_plus_covmat(dataset, pdf, covmat_t0_considered):
         X = X / rescale_fac
         pdf_cov = X @ X.T
 
+    elif pdf.error_type == 'hessian':
+        rescale_fac = pdf._rescale_factor()
+        hessian_eigenvectors = th.error_members
+        
+        # see core.HessianStats
+        X = hessian_eigenvectors[:,0::2] - hessian_eigenvectors[:,1::2]
```

Comment:
```suggestion
        X = (hessian_eigenvectors[:,0::2] - hessian_eigenvectors[:,1::2])/2.0
```

I think there's a factor of 2 missing here looking at `core.HessianStats`. When I add this I get results which are much closer between the hessian and MC PDF sets.

https://vp.nnpdf.science/kXxyXIE4SDCuIfv0jMcNHA==/

Anyway, with this I think this can be merge. If you accept the suggestion (if it works) and there's nothing else you want to change this would be done.

Reply:
I'm actually worried that there might be something else wrong because the change is not that big, but given the one that had the biggest difference in your report now looks like this:

![image](https://github.com/NNPDF/nnpdf/assets/20564015/d5f7aeee-eb7d-413e-b414-58d8de8d4b97)

(i.e., the difference between the two is even less than for nnpdf40) I guess it would be fine.

Reply:
Hi @scarlehoff, I think you are right. Having a look at eq. 5 of https://arxiv.org/pdf/1401.0013.pdf, it seems that a factor of two should be there.

> I'm actually worried that there might be something else wrong because the change is not that big,
I am not so worried about this since that might be just because the exp. uncertainty is the largest one.

I guess to check this better we would need a dataset for which the PDF uncertainty is larger/comparable to the experimental one.

Reply:
The problem is that I don't even know whether we truly have to expect that they are the same.
For now the check I've done is to ensure that the diagonal is exactly (1 std dev)**2... that happens... so the rest is probably also ok? :__